### DISCUSSION TITLE
Cascading deletes using transactions

### DISCUSSION THREAD

**Discussion by devakrishna33** [prisma/prisma#5158] [Q&A]
I wanted to use cascade on delete for some of my models, since it is not available with prisma migrate and I don't want to write postgres files to maintain the database schema, I was wondering can we use transactions for deleting all the related fields. Is there any performance limitation with respect to such implementation?

**Comment by ryands17** [comment]
@devakrishna33 👋 

If your relations are optional i.e. a `?` after the relation, then cascade delete will work fine. For e.g. in the following schema:

```prisma
model User {
  id        Int      @id @default(autoincrement())
  name      String
  posts     Post[]
}

model Post {
  id        Int      @id @default(autoincrement())
  title     String
  content   String?
  authorId  Int?
  author    User?    @relation(fields: [authorId], references: [id])
}
```

If a user is deleted, then their posts will be deleted as well. For required relations, you can use [transactions](https://www.prisma.io/docs/concepts/components/prisma-client/transactions) for deleting all the related fields. So deleting the related model first and then the main model inside the transaction should work.

It won't be a significant performance overhead because you would be doing a `deleteMany` for a relation which wouldn't take much time anyway as the primary key will always be indexed. I would suggest adding an index on the foreign key as well for a much better performance.

Let me know if you have any other queries :)

**Reply by devakrishna33** [reply]
That's all, thanks a lot

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
